TCAT/Services/Network/NetworkMonitor.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  NetworkMonitor.swift
+//  TCAT
+//
+//  Created by Jayson Hahn on 10/9/24.
+//  Copyright © 2024 Cornell AppDev. All rights reserved.
+//
+
+import Network
+import Foundation
+
+final class NetworkMonitor {
+
+    static let shared = NetworkMonitor()
+
+    private let monitor = NWPathMonitor()
+    private var status: NWPath.Status = .requiresConnection
+
+    public var isCellular: Bool = false
+    public var isReachable: Bool { status == .satisfied }
+
+    // Optional handlers for reachability changes
+    public var whenReachable: (() -> Void)?
+    public var whenUnreachable: (() -> Void)?
+
+    private init() {}
+
+    public func startMonitoring() {
+        monitor.pathUpdateHandler = { [weak self] path in
+            self?.status = path.status
+            self?.isCellular = path.isExpensive
+
+            // Notify handlers and observers based on connection status
+            if path.status == .satisfied {
+                print("Connected to the network.")
```

Comment:
should we use a logger instead of directly printing? up to you!

Reply:
My next quality code base update should come with a logger. I'll make that change later i promise.